tool/gravity/cli/config.go

```diff
@@ -851,9 +836,8 @@ func (r *agentConfig) checkAndSetDefaults() (err error) {
 	if r.token == "" {
 		return trace.BadParameter("token is required")
 	}
-	r.cloudProvider, err = validateOrDetectCloudProvider(r.cloudProvider)
-	if err != nil {
-		return trace.Wrap(err)
+	if r.cloudProvider == "" {
```

Comment:
As far as I looked, the `gravity ops agent` command that uses this config is only called by shrink operation which always sets this flag to the cluster's provider. So it should be fine to require the flag here - didn't want to over-complicate this logic for retrieving app manifest (to check default provider) if it's used only in shrink.